catalog/core/catalog-core-directorymonitor/src/main/java/org/codice/ddf/catalog/content/monitor/InputTransformerWaiter.java

```diff
@@ -0,0 +1,156 @@
+/**
+ * Copyright (c) Codice Foundation
+ *
+ * <p>This is free software: you can redistribute it and/or modify it under the terms of the GNU
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.catalog.content.monitor;
+
+import com.google.common.annotations.VisibleForTesting;
+import ddf.catalog.transform.InputTransformer;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.concurrent.TimeUnit;
+import net.jodah.failsafe.Failsafe;
+import net.jodah.failsafe.RetryPolicy;
+import org.osgi.framework.ServiceReference;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/** Waits for configured {@link InputTransformer}s. */
+public class InputTransformerWaiter {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(InputTransformerWaiter.class);
+
+  private static final String TRANSFORMER_WAIT_TIMEOUT_PROPERTY =
+      "org.codice.ddf.cdm.transformerWaitTimeoutSeconds";
+
+  private static final long DEFAULT_TRANSFORMER_WAIT_TIMEOUT_SECONDS =
+      TimeUnit.MINUTES.toSeconds(5);
+
+  private static final long DEFAULT_TRANSFORMER_CHECK_PERIOD_MILLIS = TimeUnit.SECONDS.toMillis(20);
```

Comment:
It necessary to have both configurable values in the system properties and default values in code? Might it just be enough to add them to the system properties?

Reply:
I like the default in code in case the system property is not available.